### PR TITLE
Feat : added search and alerts and other functionality

### PR DESCRIPTION
**Pull request by sashaboi**
Added search product and alerts and other small improvements .
Please review.

### CODE REVIEW

File: src/pages/components/Alert/alert.css [hunk 1]
```diff
@@ -0,0 +1,31 @@
+.alert{
+
+    width: 275px;
+    
+    padding: 0.75rem 1.25rem;
+    font-size: 1.2rem;
+}
+.info{
+    background-color: #b3ddff;
+    border: 1px solid #0077d9;
```
Comment (vikrantkode) [inline src/pages/components/Alert/alert.css:10]: dont use hardcode color

File: src/pages/components/Alert/alert.css [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+.alert-position{
+    position: fixed;
+    bottom: 40px;
+    right: 40px;
+    z-index: 9;
+}
+.error{
+    background-color: #ff9494;
+    border: 1px solid #8d0101;
+    color: #8d0101;
```
Comment (vikrantkode) [inline src/pages/components/Alert/alert.css:23]: same here

**Review by vikrantkode — APPROVED** [review]
- some suggestion given
- good to merge

**Review by krgarima — COMMENTED** [review]
Search feature is case sensitive. I should be able to search for any product even if I write in lower case.

File: src/App.js [hunk 3]
```diff
@@ -5,18 +5,22 @@ import Homepage from './pages/home/Homepage'
 import WishList from "./pages/wishlist/Wishlist";
 import Cart from "./pages/cart/Cart";
 import Products from "./pages/products/Products";
+import Product from './pages/product/Product';
 import SignUp from "./pages/auth/signup/SignUp"
 import Login from "./pages/auth/login/Login"
+import LostElement from './pages/LostElement/LostElement';
```
Comment (krgarima) [inline src/App.js:11]: You can import all of them in a single index.js file and export it here.

File: src/index.js [hunk 4]
```diff
@@ -6,18 +6,25 @@ import { makeServer } from "./server";
 import {BrowserRouter as Router} from 'react-router-dom'
 import { CartProvider } from "./pages/context/cartcontext";
 import {WishListProvider} from "./pages/context/wishlistcontext"
+import {AlertProvider} from './pages/context/AlertContext'
```
Comment (krgarima) [inline src/index.js:9]: These too.

File: src/pages/components/product-card/ProductCard.jsx [hunk 5]
```diff
@@ -1,23 +1,32 @@
...
+import {v4 as uuid} from 'uuid'
 import { useCart } from '../../context/cartcontext';
 import axios from 'axios'
 import { useState } from 'react';
-const ProductCard = ({product}) => {
+import { useAlert } from '../../context/AlertContext'
+
+const ProductCard = ({product,renderclass}) => {
+    let navigate = useNavigate();
+    console.log(renderclass);
```
Comment (krgarima) [inline src/pages/components/product-card/ProductCard.jsx]: Remove console, if not needed.

File: src/pages/components/product-card/ProductCard.jsx [hunk 6]
```diff
@@ -1,23 +1,32 @@
...
+    const  {showalert} = useAlert();
     const [disabled , setdisabled] = useState(false)
+    const [addtocartbtn , setaddtocartbtn] =useState({text:"Add to Cart",class:"addtocart"})
+    // const [cartorgoto , setcartorgoto] = useState("cart")
```
Comment (krgarima) [inline src/pages/components/product-card/ProductCard.jsx]: Remove commented code too

File: src/pages/components/product-card/ProductCard.jsx [hunk 7]
```diff
@@ -48,33 +58,43 @@ const ProductCard = ({product}) => {
             
             axios.post('/api/user/cart',producttosend,{headers : header})
             .then((response)=>{
+                showalert({text:"Item Added to Cart !",alerttype:"success"})
+                setaddtocartbtn({text:"Add One More",class:"increasequantity"})
                 console.log('incoming cart on first time add : ',response.data.cart);
```
Comment (krgarima) [inline src/pages/components/product-card/ProductCard.jsx:63]: from here too

File: src/pages/components/product-card/ProductCard.jsx [hunk 8]
```diff
@@ -88,10 +108,11 @@ const ProductCard = ({product}) => {
...
                     ${product.price}
                 </div>
-                <button disabled={disabled} className="addtocart" onClick={()=>{addtocart(product)}} >
-                
-                    Add to Cart
-                
+                <button disabled={disabled} className={addtocartbtn.class} onClick={()=>{addtocart(product)}} >
+                    {/* {console.log(cartorgoto);}
+                    {cartorgoto === "cart" ? <div>Add to Cart</div> :<div>add one more</div>}
+                    */}
```
Comment (krgarima) [inline src/pages/components/product-card/ProductCard.jsx]: here again

File: src/pages/components/product-card/productcard.css [hunk 9]
```diff
@@ -1,14 +1,17 @@
...
     width: 300px;
 }
+.horizontal img{
+    width: 500px;
+}
 .title{
     font-size: 2rem;
     display: flex;
     align-items: center;
     justify-content: center;
```
Comment (krgarima) [inline src/pages/components/product-card/productcard.css:11]: You can make a separate class for repeated code

File: src/pages/components/product-card/productcard.css [hunk 10]
```diff
@@ -20,6 +23,25 @@
...
+}
+.horizontal{
+    width: 800px;
+    
+    border-radius: 2rem;
+    background-color: rgba(128, 128, 128,0.18);
+    display: flex;
+    align-items: center;
+    justify-content: center;
+    flex-direction: row;
```
Comment (krgarima) [inline src/pages/components/product-card/productcard.css:40]: Here, again repeatition

File: src/pages/context/AlertContext.js [hunk 11]
```diff
@@ -0,0 +1,32 @@
+import { useContext, createContext,useState } from "react";
+// eslint-disable-next-line no-unused-vars
```
Comment (krgarima) [inline src/pages/context/AlertContext.js:2]: remove this, if not needed

**Review by krgarima — COMMENTED** [review]
Some things to take care of

**Pull request merged**